Frontend/ui-customer/src/Services/API.purs

```diff
@@ -1636,14 +1636,17 @@ data FlowStatus = IDLE {}
                 | WAITING_FOR_DRIVER_ASSIGNMENT { bookingId :: String , validTill :: String }
                 | RIDE_ASSIGNED { rideId :: String }
                 | PENDING_RATING { rideId :: String }
+                | RIDE_STARTED { rideId :: String, driverLocation :: Maybe LatLong, bookingId :: String, trackingUrl :: Maybe String }
+                | RIDE_PICKUP { rideId :: String, vehicleNumber :: String, driverLocation :: Maybe LatLong, bookingId :: String, otp :: String, trackingUrl :: Maybe String, fromLocation :: LatLong}
+                | DRIVER_ARRIVED { rideId :: String, bookingId :: String, driverLocation :: Maybe LatLong, trackingUrl :: Maybe String, driverArrivalTime :: Maybe String}
 
 instance makeFlowStatusReq :: RestEndpoint FlowStatusReq FlowStatusRes where
-    makeRequest reqBody headers = defaultMakeRequest GET (EP.flowStatus "") headers reqBody
+    makeRequest reqBody@(FlowStatusReq isPolling) headers = defaultMakeRequest GET (EP.flowStatus isPolling) headers reqBody
     decodeResponse = decodeJSON
     encodeRequest req = defaultEncode req
 
 derive instance genericFlowStatusReq :: Generic FlowStatusReq _
-instance standardEncodeFlowStatusReq :: StandardEncode FlowStatusReq where standardEncode (FlowStatusReq) = standardEncode {}
+instance standardEncodeFlowStatusReq :: StandardEncode FlowStatusReq where standardEncode (FlowStatusReq isPolling) = standardEncode {}
```

Comment:
no need to extract `isPolling` from req body it can be in abstract form itself like `reqBody`